diff --git a/sources/pyside2/libpyside/CMakeLists.txt b/sources/pyside2/libpyside/CMakeLists.txt
index 7b8adfc84..0243b53e8 100644
--- a/sources/pyside2/libpyside/CMakeLists.txt
+++ b/sources/pyside2/libpyside/CMakeLists.txt
@@ -101,6 +101,7 @@ set_target_properties(pyside2 PROPERTIES
                       VERSION ${BINDING_API_VERSION}
                       SOVERSION "${PYSIDE_SO_VERSION}"
                       OUTPUT_NAME "pyside2${pyside2_SUFFIX}${SHIBOKEN_PYTHON_SHARED_LIBRARY_SUFFIX}"
+                      RUNTIME_OUTPUT_DIRECTORY_RELEASE "../PySide2"
                       DEFINE_SYMBOL PYSIDE_EXPORTS)
 
 if(Qt5Core_VERSION VERSION_GREATER "5.7.1")
