diff --git a/sources/pyside2/cmake/Macros/PySideModules.cmake b/sources/pyside2/cmake/Macros/PySideModules.cmake
index dca00ec1..5d16922c 100644
--- a/sources/pyside2/cmake/Macros/PySideModules.cmake
+++ b/sources/pyside2/cmake/Macros/PySideModules.cmake
@@ -202,7 +202,7 @@ macro(create_pyside_module)
 
     # Add target to generate pyi file, which depends on the module target.
     add_custom_target("${module_NAME}_pyi" ALL
-                      COMMAND ${CMAKE_COMMAND} -E env ${ld_prefix}
+                      COMMAND
                       "${SHIBOKEN_PYTHON_INTERPRETER}"
                       "${CMAKE_CURRENT_SOURCE_DIR}/../support/generate_pyi.py" ${generate_pyi_options})
     add_dependencies("${module_NAME}_pyi" ${module_NAME})
