diff --git a/sources/shiboken2/CMakeLists.txt b/sources/shiboken2/CMakeLists.txt
index 852e14c1..6ef00c69 100644
--- a/sources/shiboken2/CMakeLists.txt
+++ b/sources/shiboken2/CMakeLists.txt
@@ -1,10 +1,11 @@
 Include(icecc.cmake)
-project(shiboken2)
-include(CheckIncludeFileCXX)
 
 cmake_minimum_required(VERSION 3.1)
 cmake_policy(VERSION 3.1)
 
+project(shiboken2)
+include(CheckIncludeFileCXX)
+
 set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/Modules/
                       ${CMAKE_MODULE_PATH})
 
