diff --git a/sources/pyside2/cmake/Macros/PySideModules.cmake b/sources/pyside2/cmake/Macros/PySideModules.cmake
index 14707f964..554039017 100644
--- a/sources/pyside2/cmake/Macros/PySideModules.cmake
+++ b/sources/pyside2/cmake/Macros/PySideModules.cmake
@@ -197,8 +197,8 @@ macro(create_pyside_module)
         endif()
     endif()
     set(generate_pyi_options ${module_NAME} --sys-path
-        "${pysidebindings_BINARY_DIR}"
-        "${SHIBOKEN_PYTHON_MODULE_DIR}")
+        "$ENV{BUILD_PREFIX}/lib/python$ENV{PY_VER}/site-packages/shiboken2"
+        "$ENV{BUILD_PREFIX}/lib/python$ENV{PY_VER}/site-packages/PySide2")
     if (QUIET_BUILD)
         list(APPEND generate_pyi_options "--quiet")
     endif()
@@ -250,4 +250,3 @@ macro(HAS_QT_MODULE var name)
         set("end_${name}" "-->" PARENT_SCOPE)
     endif()
 endmacro()
-
