diff --git a/sources/shiboken6/ApiExtractor/clangparser/compilersupport.cpp b/sources/shiboken6/ApiExtractor/clangparser/compilersupport.cpp
index 4eb1a9ee8..137b4c44a 100644
--- a/sources/shiboken6/ApiExtractor/clangparser/compilersupport.cpp
+++ b/sources/shiboken6/ApiExtractor/clangparser/compilersupport.cpp
@@ -295,6 +295,7 @@ static QString findClangBuiltInIncludesDir()
 
 static QString compilerFromCMake(const QString &defaultCompiler)
 {
+    return qEnvironmentVariable("CXX", defaultCompiler);
     if (!compilerPath().isEmpty())
         return compilerPath();
 // Added !defined(Q_OS_DARWIN) due to PYSIDE-1032
