diff --git a/sources/pyside6/plugins/designer/CMakeLists.txt b/sources/pyside6/plugins/designer/CMakeLists.txt
index f886c7f36..49bfcca2e 100644
--- a/sources/pyside6/plugins/designer/CMakeLists.txt
+++ b/sources/pyside6/plugins/designer/CMakeLists.txt
@@ -25,7 +25,7 @@ target_compile_definitions(PySidePlugin PRIVATE -DQT_NO_KEYWORDS=1)
 # For Windows we use the limited API by default
 # See ShibokenHelpers.cmake::shiboken_check_if_limited_api() which is called always
 # with default FORCE_LIMITED_API set to TRUE for building libshiboken
-if(FORCE_LIMITED_API OR WIN32)
+if(FORCE_LIMITED_API)
     target_compile_definitions(PySidePlugin PRIVATE "-DPy_LIMITED_API=0x03090000")
 endif()
 
diff --git a/sources/shiboken6/cmake/ShibokenSetup.cmake b/sources/shiboken6/cmake/ShibokenSetup.cmake
index 32823d9fa..40df10fc7 100644
--- a/sources/shiboken6/cmake/ShibokenSetup.cmake
+++ b/sources/shiboken6/cmake/ShibokenSetup.cmake
@@ -57,7 +57,7 @@ endif()
 option(FORCE_LIMITED_API "Enable the limited API." "yes")
 set(PYTHON_LIMITED_API 0)
 
-shiboken_check_if_limited_api()
+#shiboken_check_if_limited_api()
 
 if(PYTHON_LIMITED_API)
     set_limited_api()
