diff --git a/sources/pyside6/plugins/designer/CMakeLists.txt b/sources/pyside6/plugins/designer/CMakeLists.txt
index 353d526a5..d231d8258 100644
--- a/sources/pyside6/plugins/designer/CMakeLists.txt
+++ b/sources/pyside6/plugins/designer/CMakeLists.txt
@@ -22,10 +22,7 @@ target_sources(PySidePlugin PRIVATE
 
 target_compile_definitions(PySidePlugin PRIVATE -DQT_NO_KEYWORDS=1)
 
-# For Windows we use the limited API by default
-# See ShibokenHelpers.cmake::shiboken_check_if_limited_api() which is called always
-# with default FORCE_LIMITED_API set to TRUE for building libshiboken
-if(FORCE_LIMITED_API OR WIN32)
+if(FORCE_LIMITED_API)
     target_compile_definitions(PySidePlugin PRIVATE "-DPy_LIMITED_API=0x030a0000")
 endif()
 
