diff --git a/sources/pyside6/PySide6/QtCore/CMakeLists.txt b/sources/pyside6/PySide6/QtCore/CMakeLists.txt
index 1407090ca..6961fb0a2 100644
--- a/sources/pyside6/PySide6/QtCore/CMakeLists.txt
+++ b/sources/pyside6/PySide6/QtCore/CMakeLists.txt
@@ -201,8 +201,9 @@ ${QtCore_GEN_DIR}/qtcore_module_wrapper.cpp
 
 get_property(QtCore_disabled_features TARGET Qt${QT_MAJOR_VERSION}::Core
              PROPERTY QT_DISABLED_PUBLIC_FEATURES)
+message(STATUS "QtCore_disabled_features=${QtCore_disabled_features}")
 
-if("permissions" IN_LIST QtCore_disabled_features)
+if(TRUE)
     list(APPEND QtCore_DROPPED_ENTRIES QPermission
          QBluetoothPermission QCalendarPermission QCameraPermission
          QContactsPermission QLocationPermission QMicrophonePermission)
