diff --git a/sources/pyside-tools/pyside_tool.py b/sources/pyside-tools/pyside_tool.py
index 7daacc22d..b3911e23e 100644
--- a/sources/pyside-tools/pyside_tool.py
+++ b/sources/pyside-tools/pyside_tool.py
@@ -55,7 +55,7 @@ def qt_tool_wrapper(qt_tool, args, libexec=False):
     else:
         exe = pyside_dir / qt_tool
 
-    cmd = [os.fspath(exe)] + args
+    cmd = [os.fspath(qt_tool)] + args
     returncode = subprocess.call(cmd)
     if returncode != 0:
         command = ' '.join(cmd)
