diff --git a/sources/pyside-tools/pyside_tool.py b/sources/pyside-tools/pyside_tool.py
index 7daacc22d..38b01ea2e 100644
--- a/sources/pyside-tools/pyside_tool.py
+++ b/sources/pyside-tools/pyside_tool.py
@@ -55,12 +55,12 @@ def qt_tool_wrapper(qt_tool, args, libexec=False):
     else:
         exe = pyside_dir / qt_tool
 
-    cmd = [os.fspath(exe)] + args
-    returncode = subprocess.call(cmd)
-    if returncode != 0:
+    cmd = [os.fspath(qt_tool)] + args
+    cp = subprocess.run(cmd, stderr=subprocess.PIPE, text=True, errors='replace')
+    if cp.returncode != 0:
         command = ' '.join(cmd)
-        print(f"'{command}' returned {returncode}", file=sys.stderr)
-    sys.exit(returncode)
+        print(f"'{command}' returned {cp.returncode}: {cp.stderr}", file=sys.stderr)
+    sys.exit(cp.returncode)
 
 
 def pyside_script_wrapper(script_name):
